optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorCreator.java

```diff
@@ -0,0 +1,265 @@
+            ClassOutput classOutput, ClassInfo classInfo,
+            FieldInfo fieldInfo, BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        String generatedClassName = classInfo.name().prefix().toString() + ".$optaplanner$__"
+                + classInfo.name().withoutPackagePrefix() + "$__" + fieldInfo.name();
+        ClassCreator classCreator = ClassCreator
+                .builder()
+                .className(generatedClassName)
+                .interfaces(MemberAccessor.class)
+                .classOutput(classOutput)
+                .build();
+
+        GizmoMemberAccessor.MemberDescriptor member;
+        if (Modifier.isPublic(fieldInfo.flags())) {
+            member = new GizmoMemberAccessor.MemberDescriptor(fieldInfo, indexView);
+        } else {
+            addVirtualFieldGetter(classInfo, fieldInfo, transformers);
+            String methodName = getVirtualGetterName(fieldInfo.name());
+            MethodDescriptor setterDescriptor = MethodDescriptor.ofMethod(fieldInfo.declaringClass().name().toString(),
+                    getVirtualSetterName(fieldInfo.name()),
+                    "void",
+                    fieldInfo.type().name().toString());
+            MethodInfo methodInfo = MethodInfo.create(classInfo, methodName, new org.jboss.jandex.Type[] {}, fieldInfo.type(),
+                    (short) Modifier.PUBLIC);
+            member = new GizmoMemberAccessor.MemberDescriptor(fieldInfo, methodInfo, Optional.of(setterDescriptor), indexView,
+                    fieldInfo.name());
+        }
+        try {
+            GizmoMemberAccessor.defineAccessorFor(classCreator, member,
+                    (Class<? extends Annotation>) Class.forName(annotationInstance.name().toString()));
+            classCreator.close();
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    public static void generateMethodAccessor(AnnotationInstance annotationInstance, IndexView indexView,
+            ClassOutput classOutput, ClassInfo classInfo,
+            MethodInfo methodInfo, BuildProducer<BytecodeTransformerBuildItem> transformers) {
+        String generatedClassName = classInfo.name().prefix().toString() + ".$optaplanner$__"
+                + classInfo.name().withoutPackagePrefix() + "$__" + methodInfo.name();
+        ClassCreator classCreator = ClassCreator
+                .builder()
+                .className(generatedClassName)
+                .interfaces(MemberAccessor.class)
+                .classOutput(classOutput)
+                .build();
+
+        GizmoMemberAccessor.MemberDescriptor member;
+        String name;
+        MethodInfo setterMethod;
+
+        if (methodInfo.name().startsWith("get")) {
+            name = methodInfo.name().substring(3, 4).toLowerCase(Locale.ROOT) + methodInfo.name().substring(4);
+            setterMethod = classInfo.method("set" + name.substring(0, 1).toUpperCase(Locale.ROOT) + name.substring(1),
```

Comment:
This line repeats for the boolean getter too. Maybe a method that just figures out the setter's name would do it (and make the code more testable).

Reply:
Done (Refactored to two methods (one for name, one for setter)